### DISCUSSION TITLE
sf/sfdx command to retrieve all package names

### DISCUSSION THREAD

**Discussion by jxbambrick** [forcedotcom/cli#2338] [Ideas]
**Is your feature request related to a problem? Please describe.**

As part of a deployment pipeline, I use sf to download one or more change sets (up to 100 change sets at a time). The sf command will error when a change set doesn't exist, or the name is not passed correctly (due to special characters or spaces). The only way to confirm that a change set exists is to run the command "sf project retrieve start --package-name placeholder --json" and check the status code returned. This can take several minutes to run, depending on the size of a single change set. Downloading multiple large change sets (50 or more) can take up to an hour. 

There also isn't an option to retrieve a list of all the change sets in an environment. 

**What are you trying to do**
As part of a deployment pipeline the command "sf project retrieve start --package-name placeholder1 --package-name placeholder2 --package-name placeholder3...--package-name placeholder100 --json" will fail but only show the error for the first missing or misspelled change set breaking the pipeline and taking time to manually resolve. The only way to confirm all the change sets is to run the command independently for each change set, confirm that status, and add the failed change set to a list. 

- "sf project retrieve start --package-name placeholder1--json"
- "sf project retrieve start --package-name placeholder2--json"
- "sf project retrieve start --package-name placeholder3--json"...
- - "sf project retrieve start --package-name placeholder100--json"

**Describe the solution you'd like**

I would like two commands:
- returns a list of all the change sets in an environment.
- return if a change set or list of change sets exists. 

**Describe alternatives you've considered**

The only way to confirm all the change sets exist is to run the command independently for each change set, confirm that status, and add the failed change set to a list. 

- "sf project retrieve start --package-name placeholder1--json"
- "sf project retrieve start --package-name placeholder2--json"
- "sf project retrieve start --package-name placeholder3--json"...
- "sf project retrieve start --package-name placeholder100--json"

**Additional context**
![image](https://github.com/forcedotcom/cli/assets/5697044/fd8b9a45-40af-4e85-b41c-a48d5197d05c)

[a comment by an automated account is not shown]

**Comment by SWittenstein** [comment]
I'd also like this feature; we do something similar on several of my projects.
